### DISCUSSION TITLE
Slack Integration by Tenant

### DISCUSSION THREAD

**Discussion by jmarbutt** [magicbell/community#154] [Questions] [answered]
I would like the ability to create slack integration by customer, for example we have multiple tenants that have their own slack. It would be great to let them integrate their slack into notifications. 

Or would it make sense to just have a unique project per tenant? Is there are a limit of projects?

**Comment by unamashana** [comment]
Hey @jmarbutt Do you want to create multiple Slack apps, one per customer, that will then be installed in their users' workspace? So something like a B2B2B, or B2B2C usecase, or do you simply mean multiple installation of your Slack app into the many workspaces of the different customers?

**Reply by jmarbutt** [reply]
I mean a "multiple installation of your Slack app into the many workspaces of the different customers"

So **Organization 1** users can get notifications specific to them in their slack and they can connect it to their workspace. So one app, multiple workspaces.

**Reply by unamashana** [reply]
Yes, we'd be supporting this use-case. We have a beta API up if you are interested in trying it out.

**Reply by jmarbutt** [reply]
Sure I would love to see what the api is on that.

**Reply by unamashana** [reply, marked as answer]
Apology for the radio silence on this. It took us longer than we expected to get this rolled out. The good news is, we can send you over a review env to try out the new integration and APIs. They are super functional (you can create oauth workflows, or import existing slack webhooks), as well as support for multiple apps. If you are interested in trying out the preview functionality, I can send you over the link and docs. Please ping us on our Slack - https://magicbell.to/slack.

Either way, we are about a week or two away from taking this to production (your feedback on the review env will make that faster!).
